src/lib/apparmor/profiles.rb

```diff
@@ -57,25 +59,33 @@ def to_array
       a.push(pstr)
       a
     end
+
+  private
+
+    # Executes a given command
+    #
+    # For possible parameters, see Yast::Execute.locally!.
+    #
+    # @return [Boolean] true if the command finishes correctly; false otherwise
+    def execute(*args)
+      Yast::Execute.locally!(*args)
+      true
+    rescue Cheetah::ExecutionFailed
+      false
+    end
   end
 
   # Class representing a list of profiles
   class Profiles
+    include Yast::Logger
     attr_reader :prof
     def initialize
-      status_output = `/usr/sbin/aa-status --json`
-      jtext = JSON.parse(status_output)
-      h = jtext['profiles']
       @prof = {}
-      h.each do |name, status|
-        @prof[name] = Profile.new(name, status)
-      end
-      h = jtext['processes']
-      h.each do |name, pidmap|
-        pidmap.each do |p|
-          @prof[name].addPid(p['pid'])
-        end
-      end
+      status_output = command_output("/usr/sbin/aa-status", "--pretty-json")
+      log.info("aa-status output:\n#{status_output}\n")
+      jtext = JSON.parse(status_output)
```

Comment:
sadly this will still crash. If command_output return "" for failure, it result in
```
require "json"
JSON.parse("")
=> JSON::ParserError (765: unexpected token at '')
```
try yourself in irb ;)

Reply:
That would then hold true for all branches.

Reply:
Hm, I'd see this as an upstream bug; `aa-status --json` must not return nothing.

Reply:
I'd recommend to check `$?` - if something goes wrong, it will be != 0. The possible return values are documented in the `aa-status(8)` manpage, but simply displaying something like "aa-status failed to run with statuscode $?" would be good enough.

If you really think that `aa-status --json` should return a json-encoded error message: This is possible, but it would need a change in the output format (unless we can squeeze it into one of the existing json message types), which will most probably result in a new version number embedded in the json - and therefore might need an adjustment on the YaST side. That's not impossible, I just want to outline what such a small change could trigger ;-)

Reply:
Checking the return code would be right thing, imho. But as mentioned, this pr is just about porting the current code to sle-15. Having the code beautified (or corrected) should then be done in a different pr. Else we'll end up with wildly diverging coded streams.

Reply:
it already do checking error code. If it return non-zero code it raise `Cheetah::ExecutionFailed` and it is catched at https://github.com/yast/yast-apparmor/pull/46/files#diff-dd96044f1d27ed180e7fdb36528fabefR177 and return empty string instead. So it is in our error report handling. Maybe we need to check after calling that `command_output` that result is not empty. Or handle it more nicely.